force-app/main/default/classes/ContactController.cls

```diff
@@ -31,4 +31,18 @@ public with sharing class ContactController {
             LIMIT 1
         ];
     }
+
+    @AuraEnabled(Cacheable=true)
+    public static List<Contact> getContactError() {
+        List<Contact> contactList = new List<Contact>();
+        if (contactList.size() < 5) {
+            AuraHandledException e = new AuraHandledException(
```

Comment:
@pchittum I'd recommend to avoid `AuraHandledExeption` and use any kind of custom `Exception`.

In LWC, unlike Aura, there's not reason to stick to `AuraHandledExeption`, all exceptions are surfaced to the client. Using `AuraHandledExeption` requires more code with `setMessage()` and makes things harder to test. The only reason to have `AuraHandledExeption`exposed to LWC is if you were sharing a controller method with Aura and LWC.

Reply:
Yeah...or legacy methods that were formerly used for Aura components. 

I'll have a look and spend a bit more time on this today and decide what to do. I'd like to see what happens with other exceptions. If I keep `AuraHandledException` I'll make sure to comment that it is not the only way to throw an exception (and not the best way), and is only kept in this example as an example for folks inheriting legacy implementations, or some such.